jms/components/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/listener/JMSServerConnector.java

```diff
@@ -0,0 +1,188 @@
+
+/**
+ * This is a transport listener for JMS
+ */
+public class JMSServerConnector extends ListeningServerConnector {
+    private static final Log logger = LogFactory.getLog(JMSServerConnector.class.getName());
+    private CarbonMessageProcessor carbonMessageProcessor;
+    private JMSConnectionFactory jmsConnectionFactory = null;
+    private Connection connection;
+    private Session session;
+    private Destination destination;
+    private MessageConsumer messageConsumer;
+    private Properties properties;
+    private int retryInterval = 1;
+    private int maxRetryCount = 5;
+
+    public JMSServerConnector(String id) {
+        super(id);
+    }
+
+    public void init(Map<String, String> map) {
+        properties = new Properties();
+        Set<Map.Entry<String, String>> set = map.entrySet();
+        for (Map.Entry<String, String> entry : set) {
+            String mappedParameter = JMSConstants.MAPPING_PARAMETERS.get(entry.getKey());
+            if (mappedParameter != null) {
+                properties.put(mappedParameter, entry.getValue());
+            }
+        }
+
+        String retryInterval = map.get(JMSConstants.RETRY_INTERVAL);
+        if (retryInterval != null) {
+            try {
+                this.retryInterval = Integer.parseInt(retryInterval);
+            } catch (NumberFormatException ex) {
+                logger.error("Provided value for retry interval is invalid, using the default retry interval value "
+                        + retryInterval);
+            }
+        }
+
+        String maxRetryCount = map.get(JMSConstants.MAX_RETRY_COUNT);
+        if (maxRetryCount != null) {
+            try {
+                this.maxRetryCount = Integer.parseInt(maxRetryCount);
+            } catch (NumberFormatException ex) {
+                logger.error("Provided value for max retry count is invalid, using the default max retry count "
+                        + maxRetryCount);
+            }
+        }
+
+    }
+
+
+    @Override
+    public boolean bind() {
+        try {
+            createDestinationListener();
+            return true;
+        } catch (Exception e) {
+            try {
+                if (jmsConnectionFactory == null) {
+                    throw new RuntimeException("Cannot create the jms connection factory. please check the connection"
+                            + " properties and re-deploy the jms service");
+                }
+                closeAll();
+                JMSConnectionRetryHandler jmsConnectionRetryHandler = new JMSConnectionRetryHandler(this,
+                        retryInterval, maxRetryCount);
+                jmsConnectionRetryHandler.run();
+            } catch (JMSServerConnectorException e1) {
+                throw new RuntimeException(e1.getMessage(), e1);
+            }
+        }
+        return false;
+    }
+
+    @SuppressFBWarnings({"REC_CATCH_EXCEPTION"})
+    void createDestinationListener() throws JMSServerConnectorException {
```

Comment:
shouldn't the method name be something like 'createMessageListener'. Does 'createDestinationListener' give the correct meaning?